demo/time-picker-advanced-demos.html

```diff
@@ -29,8 +29,21 @@ <h3>Custom Parser And Formatter</h3>
               parseTime: function(timeString) {
                 if (timeString) {
                   const parts = /^(\d{1,2})(?::(\d{1,2}))?(?:\s(\w{2}))?$/.exec(timeString);
+                  let hours;
+
+                  if (parts) {
+                    if (parts[1] === '12' && parts[3] === 'PM') {
+                      // for 12pm noon, set hours to 12
+                      hours = 12;
+                    } else if (parts[1] === '12' && parts[3] === 'AM') {
+                      // for 12am midnight, set hours to 0
+                      hours = 0;
+                    } else {
+                      hours = parseInt(parts[1]) + (parts[3] === 'PM' ? 12 : 0);
```

Comment:
Else statement could use similar explanation comment than the other if statements.

Reply:
hi, in my opinion, you can refactor if statements like this:
if (parts) {
    if (part[1] === '12') {
         if (part[3] === 'PM') {
              hours = 12;
         }
          else if (part[3] ==='AM') {
               hours = 0;
        }
    } 
   else {
          hours = parseInt(parts[1]) + (parts[3] === 'PM' ? 12 : 0);
  }
}

just for my opinion.

Reply:
Done.

Reply:
@hongduc-phan  Done.

Reply:
thank you